src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -254,10 +246,11 @@ object RuleUtils {
             baseOutput,
             _,
             _) =>
-        val curFiles = location.allFiles
-          .map(f => FileInfo(f.getPath.toString, f.getLen, f.getModificationTime))
-
-        val (filesDeleted, filesAppended) =
+        val (filesDeleted, filesAppended) = if (index.hasSourceUpdate) {
```

Comment:
Is it guaranteed that `location.allFiles` will always be captured in `index.deletedFiles` and `index.appendedFiles`? If so, can you add a comment why? (I didn't see a check on the signature if hybrid scan is enabled).

Reply:
Good point - we cannot guarantee that if hybrid scan is enabled. We need to check the hybrid scan config here:
```if (!hybridScanEnabled && index.hasSourceUpdate)```